recipes/gaze-detection-video/gaze-detection-video.py

```diff
@@ -0,0 +1,299 @@
+"""
+Gaze Detection Video Processor using Moondream 2
+------------------------------------------------
+Read the README.md file for more information on how to use this script. Contact us in our discord for any questions if you get stuck.
+"""
+
+import torch
+import numpy as np
+import cv2
+import matplotlib.pyplot as plt
+from PIL import Image
+from transformers import AutoModelForCausalLM
+from tqdm import tqdm
+import os
+import glob
+from typing import List, Dict, Tuple, Optional
+from contextlib import contextmanager
+
+
+def initialize_model() -> Optional[AutoModelForCausalLM]:
+    """Initialize the Moondream 2 model with error handling."""
+    try:
+        print("\nInitializing Moondream 2 model...")
+        model_id = "vikhyatk/moondream2"
+        revision = "2025-01-09"  # Specify revision for stability
+
+        if torch.cuda.is_available():
+            print(f"GPU detected: {torch.cuda.get_device_name(0)}")
+            device = "cuda"
+        else:
+            print("No GPU detected, using CPU")
```

Comment:
Would also be worth checking for mps, there’s an example in moondream/torch/sample.py